### ISSUE TITLE
Consider using Typescript instead of Babel

### ISSUE THREAD

**Issue by Meir017** [SeleniumHQ/selenium-ide#612]
## 🚀 Feature Proposal

Use typescript instead of babel

**Comment by corevo** [comment]
Let me rephrase use Typescript instead of standardized Javascript (although we use decorators which I would've liked to remove), since we might still uses Babel as the loader for webpack and so forth.  

To the topic at hand, our thought is that we use Babel (and closure compiler) out of necessity rather than preference, we would much rather have no compile time at all, in fact if we could drop bundling all together we would, as it would save time during development.  

For that reason Typescript is not very helpful, and because of it I'd like to ask you to further explain **why** this change is necessary in your mind.

**Comment by Meir017** [comment]
My two main reasons are:
* Tooling - typescript has great tooling with IntelliSense and types
* Documentation - just by annotation functions you get also documentation (.d.ts files)

**Comment by corevo** [comment]
Since none of the other [Javascript projects](https://github.com/SeleniumHQ/selenium/tree/master/javascript/node/selenium-webdriver) in Selenium uses Typescript I am inclined not to decide on my own, and make sure that if a decision on this matter will be made, it would be project wide.

**Comment by corevo** [comment]
I think that the better forum for this discussion is the [Selenium developers mailing list](https://groups.google.com/forum/#!forum/selenium-developers), I'm closing this since on my own I don't want to decide on such a drastic move in the Javascript development process of the project.

@Meir017 feel free to post your insights in the mailing list and link to this issue for reference.
